.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+<!--
+
+Thanks for improving cocotb! Here are some points to make this as smooth as possible.
+Not all of them may be applicable.
+
+Most important: please explain *why* you are proposing this change.
+
+* Make sure you have read `CONTRIBUTING.md`.
```

Comment:
Can we add a link here? A normal http://...

Reply:
Note that the PR form already has such a link, here:
![image](https://user-images.githubusercontent.com/425260/76408919-9c29b280-6385-11ea-9333-dbdddfe2b224.png)

Reply:
Yes, but the whole point of the PR template is to be very explicit about the requirements for people who might not know how an open source project or GitHub works.